src/utils/cache.rs

```diff
@@ -19,12 +19,29 @@ impl ServiceCache {
         let _: () = self.set_ex(id, dest, timeout).unwrap();
     }
 
+    pub fn invalidate_pattern(&self, pattern: &String) {
+        let keys: Iter<String> = self.scan_match(pattern).unwrap();
+        let pipeline = &mut pipe();
+        for key in keys {
+            pipeline.del(key);
+        }
+        // I don't know why I have to do this .... 3 f*n hours ...
+        let con: &redis::Connection = &self.0;
+        pipeline.execute(con);
```

Comment:
I think this could be written like this as well:
```suggestion
        pipeline.execute(&self.0);
```

However, if I understand your question correctly, its "why can't I write `pipeline.execute(self)`?" (since I think the `#[database(...)]` attribute generates a `Deref` implementation). This is because of `Deref` semantics. Usually, the compiler is smart enough to figure out that you want deref `self` into its `&Self::Target` type. However, in this case it gets confused since `execute` expects a trait object:

```rust
pub fn execute(&self, con: &dyn ConnectionLike);
```

In this particular case, there is no auto-`Deref` happening because of this. You can always force a `Deref` with the `*` operator:

```suggestion
        pipeline.execute(&*self);
        // equivalent to:
        // use std::ops::Deref;
        // pipeline.execute(self.deref());
```

Note that if this were some other method that used a concrete type as a parameter, then it would Just Work™:
```rust
fn foo(&self, con: &redis::Connection);

// This should compile:
foo(self)
```